onnxruntime/core/framework/debug_node_inputs_outputs_utils.cc

```diff
@@ -491,11 +561,20 @@ void DumpNodeInputs(
             const bool is_shape_set = (dump_options.dump_flags & NodeDumpOptions::DumpFlags::Shape) != 0;
             PrintIf(is_shape_set, MakeString(" Shape: ", shape, "\n"));
 
-            if ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::InputData) != 0) {
+            if ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::InputData) != 0 || check_half_overflow) {
               tensor_metadata.name = input_defs[i]->Name();
               tensor_metadata.step = dump_context.iteration;
               tensor_metadata.consumer = node.Name() + ":" + std::to_string(i);
-              DumpTensor(dump_options, *tensor, tensor_metadata, session_state);
+
+              TensorStatisticsData tensor_statistics;
+              DumpTensor(dump_options, *tensor, tensor_metadata, tensor_statistics, session_state);
+
+              if (check_half_overflow && tensor_statistics.is_float) {
```

Comment:
In order to detect half overflow, so the whole dump analysis should be conducted on a fp32 model, right? Do we have some constraints on the use of this function, to ensure that users would not make fp16/int4 model as input by accident?

Reply:
The design is to analysis of float tensor only, it will ignore other data type safely. It is compatible with the conversion script float16.py behavior.